src/main/kotlin/com/terraformation/backend/customer/db/UserStore.kt

```diff
@@ -337,7 +337,11 @@ class UserStore(
   }
 
   fun updateGlobalRoles(userId: UserId, roles: Set<GlobalRole>) {
-    requirePermissions { updateGlobalRoles() }
+    if (roles.isEmpty()) {
+      requirePermissions { updateGlobalRoles() }
+    } else {
+      requirePermissions { updateSpecificGlobalRoles(roles) }
```

Comment:
This check should probably look at the difference between the current and desired roles, rather than at the final list of roles.

If I want to add a new role to someone, it shouldn't fail because I don't have permission to add their existing roles. And likewise, if I want to remove a role, we should check for permission on the role I'm removing, not on the roles I'm leaving in place.